controllers/default.py

```diff
@@ -610,3 +622,40 @@ def delete():
         auth.logout()  # logout user and redirect to home page
     else:
         redirect(URL("default", "user/profile"))
+
+
+def _create_access_token(data: dict, expires=None, scopes=None) -> str:
+    """
+    Helper function to create the encoded access token using
+    the provided secret and the algorithm of the LoginManager instance
+
+    Args:
+        data (dict): The data which should be stored in the token
+        expires (datetime.timedelta):  An optional timedelta in which the token expires.
+            Defaults to 15 minutes
+        scopes (Collection): Optional scopes the token user has access to.
+
+    Returns:
+        The encoded JWT with the data and the expiry. The expiry is
+        available under the 'exp' key
+    """
+
+    to_encode = data.copy()
+
+    if expires:
+        expires_in = datetime.datetime.utcnow() + expires
+    else:
+        # default to 15 minutes expiry times
+        expires_in = datetime.datetime.utcnow() + datetime.timedelta(minutes=15)
+
+    to_encode.update({"exp": expires_in})
+
+    if scopes is not None:
+        unique_scopes = set(scopes)
+        to_encode.update({"scopes": list(unique_scopes)})
+
+    algorithm = "HS256"  # normally set in constructor
+    secret = "supersecret"  # set in settings.secret
```

Comment:
Hmm, do we want to import this from the bookserver settings eventually? It looks like we'll need to have web2py running side-by-side web bookserver for a while.

Reply:
I think the natural place to set that for web2py is in models/1.py

I guess it shows that when I started out on this, yesterday it was a total experiment.  but the experiment worked more quickly than I expected.  So we have a few things to iron out before we call it production ready.

Reply:
That makes sense. It's amazing how quick this is coming together!